artichoke-backend/src/extn/core/integer/mod.rs

```diff
@@ -140,9 +143,12 @@ impl Integer {
                     Ok(vec![chr])
```

Comment:
This `Vec<u8>` is what should be allocated a a spinoso string with the proper encoding.